packages/playwright-core/src/server/recorder.ts

```diff
@@ -251,14 +251,14 @@ export class Recorder implements InstrumentationListener {
       const { file, line } = metadata.stack[0];
       let source = this._userSources.get(file);
       if (!source) {
-        source = { isRecorded: false, file, text: this._readSource(file), highlight: [], language: languageForFile(file) };
+        source = { isRecorded: false, label: file, id: file, text: this._readSource(file), highlight: [], language: languageForFile(file) };
```

Comment:
Before we used the `file` as key of the source, this has changed with the PR and it's now `id`.
The reason for that is that `file` was used internally as a displayed value (`<option>` innerHTML) and key, but e.g. for `Library` this does not work, since there is Library in each language.

Reply:
It'd be nice to keep it a file so that we could mentally switch to tabs.